install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -103,3 +103,23 @@ spec:
             severity: warning
           annotations:
             message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90%. Master nodes starved of memory could result in degraded performance of the control plane."
+    - name: master-nodes-info-memory-usage
+      rules:
+        - alert: MasterNodesInfoMemoryUsage
+          expr: |
+            ((sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) - sum(node_memory_MemFree_bytes + node_memory_Buffers_bytes + node_memory_Cached_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" ))) / sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) * 100) > 40
+          for: 10m
+          labels:
+            severity: info
+          annotations:
+            message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 40%. Master nodes starved of memory could result in degraded performance of the etcd cluster."
+    - name: master-nodes-warn-memory-usage
+      rules:
+        - alert: MasterNodesWarnMemoryUsage
+          expr: |
+            ((sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) - sum(node_memory_MemFree_bytes + node_memory_Buffers_bytes + node_memory_Cached_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" ))) / sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) * 100) > 60
+          for: 10m
```

Comment:
`10m` is too short for warnings, which are recommended to use [`for: 60m`][1].  Admins receiving the alert may take several hours or more to get to `warning` alerts in their queue, and it's distracting to have clusters chattering away with "hey, seems like a problem for your queue!" only to say "actually, never mind, fixed now!" a few minutes later.  By the time the condition has persisted for an hour, we can reasonably expect it to last until the admin has time to investigate.

[1]: https://github.com/openshift/enhancements/blob/15aebc7549095cfc7eebc26d6460defc8e3927f6/enhancements/monitoring/alerting-consistency.md#warning-alerts

Reply:
yes i agree, i will make the warning 60 mins and the critical 30 mins, ok ?